watertap/examples/flowsheets/oaro/tests/test_oaro_multi.py

```diff
@@ -0,0 +1,108 @@
+    ConcreteModel,
+    Var,
+    value,
+)
+from idaes.core import FlowsheetBlock
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.models.unit_models import Mixer, Separator, Product, Feed
+from pyomo.util.check_units import assert_units_consistent
+import watertap.property_models.NaCl_prop_pack as props
+from watertap.unit_models.reverse_osmosis_0D import ReverseOsmosis0D
+from watertap.unit_models.osmotically_assisted_reverse_osmosis_0D import (
+    OsmoticallyAssistedReverseOsmosis0D,
+)
+
+from watertap.unit_models.pressure_changer import Pump, EnergyRecoveryDevice
+from watertap.examples.flowsheets.oaro.oaro_multi import (
+    main,
+    build,
+    set_operating_conditions,
+    initialize_system,
+    solve,
+    ERDtype,
+)
+
+solver = get_solver()
+
+# -----------------------------------------------------------------------------
+class TestOAROwithTurbine:
+    @pytest.fixture(scope="class")
+    def system_frame(self):
+        m = build(number_of_stages=2, erd_type=ERDtype.pump_as_turbine)
+
+        return m
+
+    @pytest.mark.unit
+    def test_build(self, system_frame):
+        m = system_frame
+        # model set up
+        assert isinstance(m, ConcreteModel)
+        assert isinstance(m.fs, FlowsheetBlock)
+        assert isinstance(m.fs.properties, props.NaClParameterBlock)
+
+        # unit models
+        fs = m.fs
+        # assert isinstance(fs.feed, Feed)
+        # assert isinstance(fs.P1, Pump)
+        # assert isinstance(fs.P2, Pump)
+        # assert isinstance(fs.P3, Pump)
+        # assert isinstance(fs.ERD1, EnergyRecoveryDevice)
+        # assert isinstance(fs.ERD2, EnergyRecoveryDevice)
+        # assert isinstance(fs.OARO, OsmoticallyAssistedReverseOsmosis0D)
+        # assert isinstance(fs.RO, ReverseOsmosis0D)
+        # assert isinstance(fs.product, Product)
+        # assert isinstance(fs.disposal, Product)
+
+        # additional variables
+        assert isinstance(fs.volumetric_recovery, Var)
+        assert isinstance(fs.water_recovery, Var)
+
+    @pytest.mark.component
+    def test_units(self, system_frame):
+        assert_units_consistent(system_frame)
+
+    @pytest.mark.component
+    def test_set_operating_conditions(self, system_frame):
+        m = system_frame
+        set_operating_conditions(m)
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.component
+    def test_initialize_system(self, system_frame):
+        m = system_frame
+        initialize_system(m, solver=solver)
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.component
+    def test_solution(self, system_frame):
+        m = system_frame
+        solve(m, solver=solver)
+        fs = m.fs
+        assert pytest.approx(3.57530e-5, rel=1e-5) == value(
+            fs.product.flow_mass_phase_comp[0, "Liq", "NaCl"]
+        )
+        assert pytest.approx(0.0225850, rel=1e-5) == value(fs.water_recovery)
```

Comment:
Right now, you're not actually specifying a system recovery rate to achieve and instead you're getting a low recovery of ~2% based on your model specifications. I would try to change this to see what the model gets at a specified recovery, then check some of the critical values that we'd want to check in the solution.

Reply:
I might also add tests for higher number of stages with higher feed conc and recovery too.